site-packages/integralstor/remote_monitoring.py

```diff
@@ -0,0 +1,179 @@
+import os
+import shutil
+import urllib
+import json
+from integralstor_utils import config, networking
+
+
+def _get_config_file_path():
+    server_config_file_path = None
+    try:
+        config_files_dir, err = config.get_config_files_dir()
+        if err:
+            raise Exception(err)
+        server_config_file_path = '%s/remote_monitoring_servers' % config_files_dir
+        if not os.path.isfile(server_config_file_path):
+            if not os.path.exists(config_files_dir):
+                os.makedirs(config_files_dir)
+            with open(server_config_file_path, 'w') as f:
+                pass
+    except Exception, e:
+        return None, 'Error retrieving/initializing remote monitoring server configuration file : %s' % str(e)
+    else:
+        return server_config_file_path, None
+
+
+def _generate_servers_config_file(server_dict):
+    try:
+        server_config_file_path, err = _get_config_file_path()
+        if err:
+            raise Exception(err)
+        with open('/tmp/remote_monitoring_conf_file', 'w') as wfd:
+            json.dump(server_dict, wfd, indent=2)
+
+        shutil.move('/tmp/remote_monitoring_conf_file',
+                    server_config_file_path)
+    except Exception, e:
+        return False, 'Error regenerating remote monitoring configuration file : %s' % str(e)
+    else:
+        return True, None
+
+
+def get_servers():
+    servers = {}
+    try:
+        server_config_file_path, err = _get_config_file_path()
+        if err:
+            raise Exception(err)
+        with open(server_config_file_path, 'r') as f:
+            try:
+                servers = json.load(f)
+            except Exception, e:
+                servers = {}
+                with open(server_config_file_path, 'w') as wfd:
```

Comment:
I was thinking of the off chance that if while writing the json to the file, the process dies, then we have a corrupted file which is useless. So should I just return an empty dict if the json load fails? That was the file generation will anyway regenerate it with new data